pymc3/smc/smc.py

```diff
@@ -184,109 +245,180 @@ def update_weights_beta(self):
             log_weights_un = (new_beta - old_beta) * self.likelihood_logp
             log_weights = log_weights_un - logsumexp(log_weights_un)
 
-        self.log_marginal_likelihood += logsumexp(log_weights_un) - np.log(self.draws)
         self.beta = new_beta
         self.weights = np.exp(log_weights)
         # We normalize again to correct for small numerical errors that might build up
         self.weights /= self.weights.sum()
+        self.log_marginal_likelihood += logsumexp(log_weights_un) - np.log(self.draws)
 
     def resample(self):
-        """Resample particles based on importance weights."""
+        """Resample particles based on importance weights"""
         resampling_indexes = np.random.choice(
             np.arange(self.draws), size=self.draws, p=self.weights
         )
 
-        self.posterior = self.posterior[resampling_indexes]
+        self.population = self.population[resampling_indexes]
```

Comment:
isn't `posterior` more clear than population? We already use prior and likelihood.

Reply:
My reasoning was that it isn't the posterior until the very last iteration. Whereas prior and likelihood are indeed true throughout the entire sampling